### PR TITLE
helm - new module to perform helm pull

### PR DESCRIPTION
**Pull request by abikouo**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1586

##### SUMMARY

#355 
new module to manage chart downloading ``helm pull``

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

helm_pull

### CODE REVIEW

**Comment by goneri** [comment]
recheck

**Review by gravesm — COMMENTED** [review]
A few small things, but otherwise looks good.

File: plugins/modules/helm_pull.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+    helm_pull_opt_versionning = dict(
+        skip_tls_certs_check="3.3.0",
+        chart_ca_cert="3.1.0",
+        chart_ssl_cert_file="3.1.0",
+        chart_ssl_key_file="3.1.0",
+    )
+
+    helm_pull_opts = []
+    for k, v in iteritems(helm_pull_mapping):
```
[the remark this answers is not in the document]
Comment (abikouo) [reply plugins/modules/helm_pull.py:281]: will do, thanks

File: plugins/modules/helm_pull.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        if module.params.get(k) is not None and module.params.get(k) != argspec.get(
```
[the remark this answers is not in the document]
Comment (abikouo) [reply plugins/modules/helm_pull.py:282]: some parameters are boolean, so this means the corresponding flag should be passed to the command line only if the ansible task is called with a value different from the default value.

File: plugins/modules/helm_pull.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                opt_req_version
+            ):
+                module.fail_json(
+                    msg="Parameter {0} requires helm >= {1}, current version is {2}".format(
+                        k, opt_req_version, helm_version
+                    )
+                )
+            argtype = argspec.get(k).get("type", "str")
+            if argtype == "bool":
+                if module.params.get(k) != argspec.get(k).get("default", False):
```
[the remark this answers is not in the document]
Comment (abikouo) [reply plugins/modules/helm_pull.py:296]: This comparison is useless as we already did the same above

File: plugins/modules/helm_pull.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        verify_chart=dict(key="verify"),
+        provenance=dict(key="prov"),
+        pass_credentials=dict(key="pass-credentials"),
+        skip_tls_certs_check=dict(key="insecure-skip-tls-verify"),
+        chart_devel=dict(key="devel"),
+        untar_chart=dict(key="untar"),
+    )
+
+    for k, v in helm_flag_args.items():
+        if module.params.get(k) and module.params.get(k) != v.get("default", False):
```
Comment (gravesm) [inline plugins/modules/helm_pull.py:302]: This is still broken. If the default value were True and the user passed in True, or just left it unset, it's going to not add the flag. Can we just remove any comparison to a default value at all? I don't think there's any need for it, and it is adding unnecessary complexity. These are boolean values that control whether or not a flag is added to the command line. All we care about is whether the value we get in module.params is True or False. If it's True we set the flag, if it's False we don't.

Comment (abikouo) [reply plugins/modules/helm_pull.py]: updated

File: plugins/modules/helm_pull.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+    - Requires helm >= 3.1.0.
+    type: path
+  chart_ssl_key_file:
+    description:
+    - Identify HTTPS client using this SSL key file
+    - Requires helm >= 3.1.0.
+    type: path
+
+extends_documentation_fragment:
+  - kubernetes.core.helm_common_options
```
Comment (gravesm) [inline plugins/modules/helm_pull.py:113]: Does this actually use any of these options? Unless I'm missing something, it doesn't seem like it's necessary.

**Review by gravesm — COMMENTED** [review]
Sorry for taking so long to get to this. I think the module params just need to be cleaned up.

File: plugins/modules/helm_pull.py (hunk 5 above)
Comment (abikouo) [reply plugins/modules/helm_pull.py]: Removed.

**Review by gravesm — APPROVED** [review]

**Pull request merged**